src/realm/query_expression.hpp

```diff
@@ -345,6 +345,11 @@ class ValueBase {
     {
         return m_first + m_size;
     }
+    void truncate(ValueType* from)
+    {
+        auto new_size = from - m_first;
+        m_size = new_size;
```

Comment:
if truncate crosses the threshold from the > 8 to 8 elements (size of the prealloc array) we need to deallocate the dynamic memory. I think we can just use the existing `resize(size_t)` method instead?
```

1: =================================================================
1: ==1751==ERROR: LeakSanitizer: detected memory leaks
1: 
1: Direct leak of 30256 byte(s) in 122 object(s) allocated from:
1:     #0 0x5ca85d in operator new[](unsigned long) (/mnt/jenkins/workspace/realm_realm-core_PR-5678/build-dir/test/realm-tests+0x5ca85d)
1:     #1 0x753c0e in realm::ValueBase::resize(unsigned long) /mnt/jenkins/workspace/realm_realm-core_PR-5678/src/realm/query_expression.hpp:681:27
1: 
1: SUMMARY: AddressSanitizer: 30256 byte(s) leaked in 122 allocation(s).
```

Reply:
Sure - this has to be handled